Analysis/Tasks/PWGHF/HFLcCandidateSelector.cxx

```diff
@@ -328,27 +332,41 @@ struct HFLcCandidateSelector {
         continue;
       }
 
-      proton = selectionPID(trackPos1, 2212);
+      proton1 = selectionPID(trackPos1, 2212);
+      proton2 = selectionPID(trackPos2, 2212);
       kaonMinus = selectionPID(trackNeg1, 321);
-      pionPlus = selectionPID(trackPos2, 211);
+      pionPlus1 = selectionPID(trackPos1, 211);
+      pionPlus2 = selectionPID(trackPos2, 211);
 
-      if (proton == 0 || kaonMinus == 0 || pionPlus == 0) {
-        pidLc = 0; //exclude Lc
-      }
-      if (proton == 1 && kaonMinus == 1 && pionPlus == 1) {
-        pidLc = 1; //accept Lc
+      if (!d_FilterPID) {
+        // PID non applied
+        pidLcpKpi = 1;
+        pidLcpiKp = 1;
+      } else {
```

Comment:
It must be sad to call `selectionPID` five times and then realise that `!d_FilterPID`. :'-(

Reply:
Good Point! I can move them inside the else. Thanks!